server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -428,12 +423,23 @@ public static RepositoryData snapshotsFromXContent(final XContentParser parser,
                                         // since we already have the name/uuid combo in the snapshots array
                                         uuid = parser.text();
                                     }
-                                    snapshotIds.add(snapshots.get(uuid));
+                                    if (uuid != null) {
```

Comment:
should this always be non-null? Should we barf if not?

Reply:
It should be non null, but I removed the null check so that it will throw the same exception.